modules/diskq/logqueue-disk-reliable.c

```diff
@@ -296,11 +312,16 @@ _push_tail(LogQueueDisk *s, LogMessage *msg, LogPathOptions *local_options, cons
       g_queue_push_tail (self->qreliable, LOG_PATH_OPTIONS_TO_POINTER(path_options));
       log_msg_ref (msg);
 
-      log_queue_memory_usage_add(&self->super.super, log_msg_get_size(msg));
-      local_options->ack_needed = FALSE;
+      log_queue_memory_usage_add(s, log_msg_get_size(msg));
+      local_options.ack_needed = FALSE;
     }
 
-  return TRUE;
+  log_queue_push_notify(s);
+  log_queue_queued_messages_inc(s);
+  log_msg_ack(msg, &local_options, AT_PROCESSED);
+  log_msg_unref(msg);
```

Comment:
Just a note: Now that this has been moved here, it can be seen that `log_msg_ack()` should not always be called, that's why we have the `ack_needed = FALSE;` trickery a few lines above.